submissions/shramkoweb/memory-game/main.js

```diff
@@ -0,0 +1,118 @@
+const HIDE_TIMEOUT = 500;
+const RESET_TIMEOUT = 700;
+const SHOW_ALL_CARDS_TIMEOUT = 1000;
+const CARDS_COUNT_PER_FLIP = 2;
+
+const contentElement = document.querySelector('.content');
+const cellElements = contentElement.querySelectorAll('.cell');
+const resultElement = document.querySelector('.result');
+
+const state = {
+    isBoardBlocked: false,
+    firstCard: null,
+    secondCard: null,
+    cardsLeft: cellElements.length,
+};
+
+const resetBoardState = () => {
+    state.secondCard = null;
+    state.firstCard = null;
+    state.isBoardBlocked = false;
+};
+
+const hideAllCards = () => {
+    state.isBoardBlocked = false;
+    cellElements.forEach(card => {
+        card.classList.remove('opened');
+    });
+};
+
+const getOrder = (cardsCount) => String(Math.floor(Math.random() * cardsCount));
+
+const temporaryShowAllCards = () => {
+    state.isBoardBlocked = true;
+    cellElements.forEach(card => {
+        card.classList.add('opened');
+    });
+
+    setTimeout(hideAllCards, SHOW_ALL_CARDS_TIMEOUT);
+};
+
+const flipCard = (card) => {
+    const isClickOnEqualCard = card === state.firstCard;
+
+    if (
+        state.isBoardBlocked ||
+        isClickOnEqualCard ||
+        !card
+    ) {
+        return;
+    }
+    card.classList.add('flip', 'opened');
+
+    if (!state.firstCard) {
+        state.firstCard = card;
+    } else {
+        state.secondCard = card;
+        checkIsCardsMatch();
+    }
+};
+
+const isGameWin = () => {
+    console.log(state);
```

Comment:
Remove this line and you have my approval.

Reply:
lol =)